### PR TITLE
Fix turn into Invalid direction

### LINKED ISSUE #4349: `Direction.TurnCw`/`TurnCcw` are bugged
image

TurnCw from south becomes invalid

### PR DESCRIPTION
**Pull request by Morb0**
Fixes #4349

### CODE REVIEW

File: Robust.Client/Utility/DirExt.cs [hunk 1]
```diff
@@ -103,7 +103,7 @@ public static RSIDirection Convert(this Direction dir, RSI.State.DirectionType t
 
         public static Direction TurnCw(this Direction dir)
         {
-            return (Direction)(((int)dir - 1) % 8);
+            return (Direction)(((int)dir - 1 + 8) % 8);
         }
```
Comment (metalgearsloth) [inline Robust.Client/Utility/DirExt.cs]: Just do +7, also please make a test for it.

Comment (Morb0) [reply Robust.Client/Utility/DirExt.cs]: For me correctly work only +8

Comment (metalgearsloth) [reply Robust.Client/Utility/DirExt.cs]: do you mean - 1 + 8 or just doing + 7?

Comment (Morb0) [reply Robust.Client/Utility/DirExt.cs]: I mean `(Direction)(((int)dir - 1 + 7) % 8)`

Comment (ElectroJr) [reply Robust.Client/Utility/DirExt.cs]: I'm fairly sure sloth meant you should replace `- 1 + 8` with `+ 7`

Comment (Morb0) [reply Robust.Client/Utility/DirExt.cs]: Okay, make sense

**Pull request merged**